### PR TITLE
Added Local Network Gateway API

### PR DESCRIPTION
**Pull request by nacx**
This PR adds the Local Gateway Network API and is the first of a set of PRs that will add support for configuring VPNs in Azure ARM: https://docs.microsoft.com/en-us/rest/api/network-gateway/

### CODE REVIEW

**Comment by nacx** [comment]
Live test result:
```bash
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running org.jclouds.azurecompute.arm.features.LocalNetworkGatewayApiLiveTest
Configuring TestNG with: TestNG652Configurator
Starting test createLocalNetworkGateway(org.jclouds.azurecompute.arm.features.LocalNetworkGatewayApiLiveTest)
[TestNG] Test createLocalNetworkGateway(org.jclouds.azurecompute.arm.features.LocalNetworkGatewayApiLiveTest) succeeded: 1830ms
Test suite progress: tests succeeded: 1, failed: 0, skipped: 0.
Starting test getLocalNetworkGateway(org.jclouds.azurecompute.arm.features.LocalNetworkGatewayApiLiveTest)
[TestNG] Test getLocalNetworkGateway(org.jclouds.azurecompute.arm.features.LocalNetworkGatewayApiLiveTest) succeeded: 304ms
Test suite progress: tests succeeded: 2, failed: 0, skipped: 0.
Starting test listLocalNetworkGateways(org.jclouds.azurecompute.arm.features.LocalNetworkGatewayApiLiveTest)
[TestNG] Test listLocalNetworkGateways(org.jclouds.azurecompute.arm.features.LocalNetworkGatewayApiLiveTest) succeeded: 333ms
Test suite progress: tests succeeded: 3, failed: 0, skipped: 0.
Starting test updateLocalNetworkGateway(org.jclouds.azurecompute.arm.features.LocalNetworkGatewayApiLiveTest)
[TestNG] Test updateLocalNetworkGateway(org.jclouds.azurecompute.arm.features.LocalNetworkGatewayApiLiveTest) succeeded: 16668ms
Test suite progress: tests succeeded: 4, failed: 0, skipped: 0.
Starting test deleteLocalNetworkGateway(org.jclouds.azurecompute.arm.features.LocalNetworkGatewayApiLiveTest)
[TestNG] Test deleteLocalNetworkGateway(org.jclouds.azurecompute.arm.features.LocalNetworkGatewayApiLiveTest) succeeded: 13891ms
Test suite progress: tests succeeded: 5, failed: 0, skipped: 0.
Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 76.254 sec - in org.jclouds.azurecompute.arm.features.LocalNetworkGatewayApiLiveTest

Results :

Tests run: 5, Failures: 0, Errors: 0, Skipped: 0
```

File: providers/azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/AvailabilitySetApiMockTest.java [hunk 1]
```diff
@@ -138,7 +138,7 @@ public void deleteAvailabilitySet() throws InterruptedException {
       assertSent(server, "DELETE", path);
    }
 
-   public void deleteSubnetResourceDoesNotExist() throws InterruptedException {
+   public void deleteAvailabilitySetDoesNotExist() throws InterruptedException {
```
Comment (andreaturli) [inline providers/azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/AvailabilitySetApiMockTest.java]: thanks for fixing it!

File: providers/azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/LocalNetworkGatewayApiLiveTest.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+   private LocalNetworkGatewayApi api() {
+      return api.getLocalNetworkGatewayApi(resourceGroupName);
+   }
+
+   private boolean waitUntilAvailable(final String name) {
+      return resourceAvailable.apply(new Supplier<Provisionable>() {
+         @Override
+         public Provisionable get() {
+            LocalNetworkGateway gw = api().get(name);
+            return gw == null ? null : gw.properties();
```
Comment (andreaturli) [inline providers/azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/LocalNetworkGatewayApiLiveTest.java]: why not using `provisioningState` ?

Comment (nacx) [reply providers/azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/LocalNetworkGatewayApiLiveTest.java]: That's actually what it does. The predicate accepts a `Provisionable` object which is just an interface that exposes the `provisioningState` and then uses that state to check the desired status.

**Pull request merged**